Design.md

```diff
@@ -0,0 +1,128 @@
+# Job Worker Design Doc
+
+Design doc for a prototype job worker service that provides an API to run arbitrary Linux processes.
+
+## Level 2 requirements
+
+- Library: Worker library with methods to start/stop/query status and get the output of a job.
+- API:
+    - HTTPS API to start/stop/get status of a running process.
+    - Use mTLS authentication and verify client certificate. Set up strong set of cipher suites for TLS and good crypto
+      setup for certificates. Do not use any other authentication protocols on top of mTLS.
+    - Use a simple authorization scheme.
+- CLI:
+    - CLI should be able to connect to worker service and schedule several jobs.
+    - CLI should be able to query result of the job execution and fetch its output.
+
+## Job object
+
+- `ID` unique identifier of a job
+- `Command` linux command to be executed
+- `Status` current status of the job
+- `Output` output of the job process
+- `Error` exit error or, termination error
```

Comment:
What will `Error` be? A string, integer, something else?

Reply:
A string